code/controllers/CommentingController.php

```diff
@@ -241,12 +241,18 @@ public function spam() {
 			return Security::permissionFailure($this, 'You do not have permission to edit this comment');
 		}
 		if(!$comment->getSecurityToken()->checkRequest($this->request)) return $this->httpError(400);
-		
-		$comment->markSpam();
 
-		return $this->request->isAjax()
-			? $comment->renderWith('CommentsInterface_singlecomment')
-			: $this->redirectBack();
+		$comment->markSpam();
+        if (empty($this->getRequest()->getHeader('Referer'))) {
```

Comment:
In fact, you should probably put the if(is_ajax) on the outer loop, to prevent duplication of the ajax response handler which doesn't care about referer.